lib/services/user_service.ex

```diff
@@ -12,7 +12,18 @@ defmodule SlackCoder.Services.UserService do
     if db_user, do: {:ok, user}, else: {:new, user}
   end
 
+  def save(%Ecto.Changeset{data: user} = changeset) do
+    pre_save(changeset.changes, user)
+    user = Repo.update!(changeset)
+    SlackCoder.Users.Supervisor.start_user(user)
+    user
```

Comment:
Your controller is expecting `{:ok, user}`... So what you should do is change this into a `case` and do `Repo.insert_or_update` to make it more reusable (and remove the `!`). Then you won't feel so bad hard coding an "ok" return tuple.